### PR TITLE
Move settings out of model

### PR DESCRIPTION
**Pull request by CharliePoole**
Continuing isolation of model... partial fix to #189

### CODE REVIEW

File: src/nunit.gui.core/Model/TestModel.cs [hunk 1]
```diff
@@ -202,12 +198,12 @@ public void NewProject()
 
         public void NewProject(string filename)
         {
-            MessageBox.Show("It's not yet decided if we will implement saving of projects. The alternative is to require use of the project editor, which already supports this.", "Not Yet Implemented");
+            throw new NotImplementedException();
```
Comment (immeraufdemhund) [inline src/nunit.gui.core/Model/TestModel.cs]: Why this change? I understand the gist of it, but I don't see where we are catching this type of exception to show the user. So when this is called by the MainPresenter would this cause an unhandled exception to occur. Same for `SaveProject`

File: src/tests/Presenters/UserSettingsFake.cs [hunk 2]
```diff
@@ -28,6 +28,9 @@
 
 namespace NUnit.Gui.Presenters
 {
+    // TODO: With latest changes, this is no longer used.
+    // Keeping it for a while, as we may end up wanting
+    // to write more tests that use it.
     class UserSettingsFake : ISettings
```
Comment (immeraufdemhund) [inline src/tests/Presenters/UserSettingsFake.cs]: Maybe consider marking this as obsolete with the same message, then either delete it or flip the flag to make it cause a compile error.

File: src/nunit.gui.core/Model/TestModel.cs (hunk 1 above)
Comment (CharliePoole) [reply src/nunit.gui.core/Model/TestModel.cs]: Good catch. I intended to move the message display to the presenter. This gets System.Windows.Forms out if the model.

File: src/tests/Presenters/UserSettingsFake.cs (hunk 2 above)
Comment (CharliePoole) [reply src/tests/Presenters/UserSettingsFake.cs]: Since it's in our tests, I see no harm in it for now. This is part of a series of changes to the same code so I'll be coming back here soon.
